mlrun/projects/project.py

```diff
@@ -3030,6 +3031,27 @@ def list_runs(
             **kwargs,
         )
 
+    def register_datastore_profile(self, profile: DatastoreProfile):
+        project_ds_name_private = DatastoreProfile.generate_secret_key(
+            profile.name, self.name
+        )
+        private_body = DatastoreProfile2Json.get_json_private(profile)
+        public_body = DatastoreProfile2Json.get_json_public(profile)
+        # set project public data to DB
+        public_profile = mlrun.common.schemas.DatastoreProfile(
+            name=profile.name, type=profile.type, body=public_body, project=self.name
+        )
+        mlrun.db.get_run_db(secrets=self._secrets).store_datastore_profile(
+            public_profile, self.name
+        )
+        # Set local environment variable
+        environ[project_ds_name_private] = private_body
+        # set project secret
+        self.set_secrets(
+            secrets={project_ds_name_private: private_body},
+            provider=mlrun.common.schemas.SecretProviderName.kubernetes,
+        )
+
```

Comment:
A point to consider - you don't register the dsp in the project object itself, which means that if you export/import the project then the dsp will not be migrated with the project. Is this intentional? We need to think whether we want dsps to be migrated with projects (without secrets, obviously).

Reply:
Honestly, I hadn't thought about it before. I'll let you decide on that